src/timeseriesflattenerv2/spec_processors/temporal.py

```diff
@@ -53,34 +80,46 @@ def _mask_outside_lookperiod(
     else:
         lookperiod_string = f"{lookperiod.first.days}_to_{lookperiod.last.days}_days"
 
-    new_colname = f"{column_prefix}_{value_col_name}_within_{lookperiod_string}"
+    # TODO: #436 base suffix on the type of timedelta (days, hours, minutes)
+    new_colnames = [
+        f"{column_prefix}_{value_col_name}_within_{lookperiod_string}"
+        for value_col_name in value_col_names
+    ]
 
     return TimeMaskedFrame(
-        init_df=masked_frame.rename({timedelta_frame.value_col_name: new_colname}),
+        init_df=masked_frame.rename(dict(zip(value_col_names, new_colnames))),
         pred_time_uuid_col_name=timedelta_frame.pred_time_uuid_col_name,
-        value_col_name=new_colname,
+        value_col_names=new_colnames,
         timestamp_col_name=timedelta_frame.value_timestamp_col_name,
     )
 
 
 def _aggregate_masked_frame(
     masked_frame: TimeMaskedFrame, aggregators: Sequence[Aggregator], fallback: ValueType
 ) -> pl.LazyFrame:
-    aggregator_expressions = [aggregator(masked_frame.value_col_name) for aggregator in aggregators]
+    aggregator_expressions = [
+        aggregator(value_col_name)
+        for aggregator in aggregators
+        for value_col_name in masked_frame.value_col_names
+    ]
 
     grouped_frame = masked_frame.init_df.group_by(
         masked_frame.pred_time_uuid_col_name, maintain_order=True
     ).agg(aggregator_expressions)
 
     value_columns = (
         Iter(grouped_frame.columns)
-        .filter(lambda col: masked_frame.value_col_name in col)
+        .filter(
+            lambda col: any(
+                value_col_name in col for value_col_name in masked_frame.value_col_names
+            )
+        )
         .map(lambda old_name: (old_name, f"{old_name}_fallback_{fallback}"))
     )
     rename_mapping = dict(value_columns)
 
     with_fallback = grouped_frame.with_columns(
-        cs.contains(masked_frame.value_col_name).fill_null(fallback)
+        cs.contains(masked_frame.value_col_names).fill_null(fallback)
```

Comment:
I have a hard time reasoning well about this code, and AFAICT, it's untested. We need a test with multiple input columns and multiple aggregators 😊

Reply:
Added, it works 🙌